hca/full_upload.py

```diff
@@ -0,0 +1,177 @@
+                    filenames.append(filename)
+                    files_to_upload.append(open(full_file_name, "rb"))
+            else:  # It's a file
+                filenames.append(os.path.basename(path))
+                files_to_upload.append(open(path, "rb"))
+
+        # Print to stderr, upload the files to s3 and return a list of tuples: (filename, filekey)
+        eprint("Uploading the following keys to aws:")
+        for file in filenames:
+            eprint("\t", file)
+
+        uploaded_keys = upload_to_cloud(files_to_upload, args["staging_bucket"], args["replica"])
+
+        filename_key_list = list(zip(filenames, uploaded_keys))
+        eprint("\nThe following keys were uploaded successfuly:")
+        for filename, key in filename_key_list:
+            eprint('\t{:<12}  {:<12}'.format(filename, key))
+        return filename_key_list
+
+    @classmethod
+    def _put_files(cls, filename_key_list, staging_bucket, api):
+        """Use the API class to make a put-files request on each of these files."""
+        bundle_uuid = str(uuid.uuid4())
+        files = []
+        for (filename, key) in filename_key_list:
+            eprint("File {}: registering...".format(filename))
+
+            # Generating file data
+            creator_uid = os.environ.get(cls.CREATOR_ID_ENVIRONMENT_VARIABLE, "1")
+            source_url = "s3://{}/{}".format(staging_bucket, key)
+            file_uuid = key[:key.find("/")]
+            eprint("File {}: assigned uuid {}".format(filename, file_uuid))
+
+            response = api.make_request([
+                "put-files",
+                file_uuid,
+                "--bundle-uuid", bundle_uuid,
+                "--creator-uid", creator_uid,
+                "--source-url", source_url
+            ], stream=True)
+
+            if response.ok:
+                version = response.json().get("version", "blank")
+                eprint("File {}: registered with uuid {}".format(filename, file_uuid))
+                files.append({
+                    "name": filename,
+                    "version": version,
+                    "uuid": file_uuid,
+                    "creator_uid": creator_uid
+                })
+                response.close()
+
+            else:
+                eprint("File {}: registration FAILED".format(filename))
+                eprint(response.text)
+                response.close()
+                response.raise_for_status()
+
+            eprint("Request response")
+            eprint("{}".format(response.content.decode()))
+        return bundle_uuid, files
+
+    @classmethod
+    def _put_bundle(cls, bundle_uuid, files, api):
+        """Use the API class to make a put-bundles request."""
+        file_args = [Constants.OBJECT_SPLITTER.join(["True", file["name"], file["uuid"], file["version"]]) for file in files]
+        creator_uid = os.environ.get(cls.CREATOR_ID_ENVIRONMENT_VARIABLE, "1")
+        replica = "aws"
```

Comment:
Yeah definitely. Noticed and changed this in download, forgot to do the same for this one.